### PR TITLE
fix: Clean up Docker for Release, Modify Pipeline with Release Logic, Fix Binary Links / Version from recent Change.fix

### PR DESCRIPTION
**Pull request by gzukel**
# Description

**This PR includes the following:**

1. Cleaned up Dockerfile called Dockerfile-Release that uses a builder image to do the building and then copies the binary into a smaller new image that doesn't have the build trash.
2. Modified the release pipeline to only run on PR merge to main.
3. Updated the binary link names in the rpc pipelines from the recent release and binary versions.

Closes: <PD-XXXX>

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

I need to merge this, then merge to main to test to ensure things work properly. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [x] Tested via GitHub Actions 

# Checklist:

- [N/A] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -80,3 +80,14 @@ to [run the smoke test](./contrib/localnet/README.md).
 [Twitter](https://twitter.com/zetablockchain) |
 [Discord](https://discord.com/invite/zetachain) |
 [Telegram](https://t.me/zetachainofficial) | [Website](https://zetachain.com)
+
+## Creating a Release for Mainnet
+Creating a release for Mainnet is a straightforward process. Here are the steps to follow:
+
+#### Steps
```
Comment (lumtis) [inline readme.md:87]:
```suggestion
### Steps
```

File: readme.md [hunk 2]
```diff
@@ -80,3 +80,14 @@ to [run the smoke test](./contrib/localnet/README.md).
 [Twitter](https://twitter.com/zetablockchain) |
 [Discord](https://discord.com/invite/zetachain) |
 [Telegram](https://t.me/zetachainofficial) | [Website](https://zetachain.com)
+
+## Creating a Release for Mainnet
+Creating a release for Mainnet is a straightforward process. Here are the steps to follow:
```
Comment (lumtis) [inline readme.md:85]:
```suggestion
Creating a release for mainnet is a straightforward process. Here are the steps to follow:
```

File: readme.md [hunk 3]
```diff
@@ -80,3 +80,14 @@ to [run the smoke test](./contrib/localnet/README.md).
 [Twitter](https://twitter.com/zetablockchain) |
 [Discord](https://discord.com/invite/zetachain) |
 [Telegram](https://t.me/zetachainofficial) | [Website](https://zetachain.com)
+
+## Creating a Release for Mainnet
```
Comment (lumtis) [inline readme.md:84]: Shouldn't we also have instructions for testnet release?

**Review by lumtis — COMMENTED** [review]
Looks good to me overall.
It's mainly DevOps related, maybe we can try having 2 approvals from @zeta-chain/devops for this PR

**Review by julianrubino — APPROVED** [review]

**Review by CharlieMc0 — APPROVED** [review]

**Review by lumtis — APPROVED** [review]

**Pull request merged**